src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,223 @@
+
+            uint256 startCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+            // transfer CONVEX tokens from CrvRewards back to the bridge
+            ICurveRewards(selectedPool.curveRewards).withdraw(_totalInputValue, true); // always claim rewards
+
+            BOOSTER.withdraw(selectedPool.poolId, _totalInputValue);
+
+            uint256 endCurveLpTokens = ICurveLpToken(_outputAssetA.erc20Address).balanceOf(address(this));
+
+            outputValueA = (endCurveLpTokens - startCurveLpTokens);
+
+            IRepConvexToken(_inputAssetA.erc20Address).burn(_totalInputValue);
+        } else {
+            revert ErrorLib.InvalidOutputA();
+        }
+
+        // Pay out subsidy to the rollupBeneficiary
+        SUBSIDY.claimSubsidy(
+            computeCriteria(_inputAssetA, _inputAssetB, _outputAssetA, _outputAssetB, 0),
+            _rollupBeneficiary
+        );
+    }
+
+    /**
+    @notice Loads pool information for all pools supported by Convex Finance.
+    @notice Set allowance for Rollup's Curve LP tokens.
+    @notice Cached. Loads only new pools.
+    */
+    function loadPools() public {
+        uint256 currentPoolLength = BOOSTER.poolLength();
+        if (currentPoolLength != poolLength) {
+            for (uint256 i = poolLength; i < currentPoolLength; i++) {
+                (address curveLpToken, address convexToken, , address curveRewards, , ) = BOOSTER.poolInfo(i);
+                pools[curveLpToken] = PoolInfo(uint96(i), convexToken, curveRewards);
+
+                // deploy token, log token address
+                address deployedToken = address(new RepresentingConvexToken("RepresentingConvexToken", "RCT"));
+                // deployedTokens[curveLpToken] = address(new RepresentingConvexToken(string.concat("RepresentingConvexToken", Strings.toString(i)), string.concat("RCT", Strings.toString(i))));
+                deployedTokens[curveLpToken] = deployedToken;
+                // approvals
+                ICurveLpToken(curveLpToken).approve(address(BOOSTER), type(uint256).max);
+                ICurveLpToken(curveLpToken).approve(ROLLUP_PROCESSOR, type(uint256).max);
+                IRepConvexToken(deployedToken).approve(ROLLUP_PROCESSOR, type(uint256).max);
+            }
+            poolLength = currentPoolLength;
+        }
+    }
+
+    /**
+     * @notice Computes the criteria that is passed when claiming subsidy.
+     * @param _inputAssetA The input asset
+     * @param _outputAssetA The output asset
+     * @return The criteria
+     */
+    function computeCriteria(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint64
+    ) public view override(BridgeBase) returns (uint256) {
+        if (deployedTokens[_inputAssetA.erc20Address] == _outputAssetA.erc20Address) {
```

Comment:
Both branches are computing the same thing.

Reply:
Resolved